packages/superset-ui-core/src/query/extractQueryFields.ts

```diff
@@ -46,26 +45,45 @@ export default function extractQueryFields(
   };
   const finalQueryFields: QueryFields = {
     columns: [],
-    groupby: [],
     metrics: [],
   };
+  const { query_mode: queryMode, include_time: includeTime, ...restFormData } = formData;
 
-  Object.entries(formData).forEach(([key, value]) => {
-    const normalizedKey = queryFieldAliases[key] || key;
-    if (normalizedKey in finalQueryFields) {
-      if (normalizedKey === 'metrics') {
-        finalQueryFields[normalizedKey] = finalQueryFields[normalizedKey].concat(value);
-      } else {
-        // currently the groupby and columns field only accept pre-defined columns (string shortcut)
-        finalQueryFields[normalizedKey] = finalQueryFields[normalizedKey]
-          .concat(value)
-          .filter(x => typeof x === 'string' && x);
+  Object.entries(restFormData).forEach(([key, value]) => {
+    // ignore `null` or `undefined` value
+    if (value == null) {
+      return;
+    }
+
+    let normalizedKey: string = queryFieldAliases[key] || key;
+
+    // ignore groupby when in raw records mode
+    if (queryMode === QueryMode.raw && normalizedKey === 'groupby') {
+      return;
+    }
+    if (queryMode === QueryMode.aggregate) {
+      // ignore columns when (specifically) in aggregate mode.
+      if (normalizedKey === 'columns') {
```

Comment:
can we unit these 2 conditions? like:
```
if (queryMode === QueryMode.aggregate && normalizedKey === 'columns) {
```

Reply:
Sure. There was more conditions in this branch before but I must have cleaned them up.